MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -251,15 +251,23 @@ public Task SaveImage(BaseItem item, Stream source, string mimeType, ImageType t
         }
 
         /// <inheritdoc/>
-        public Task SaveImage(BaseItem item, string source, string mimeType, ImageType type, int? imageIndex, bool? saveLocallyWithMedia, CancellationToken cancellationToken)
+        public async Task SaveImage(BaseItem item, string source, string mimeType, ImageType type, int? imageIndex, bool? saveLocallyWithMedia, CancellationToken cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(source))
             {
                 throw new ArgumentNullException(nameof(source));
             }
 
             var fileStream = AsyncFile.OpenRead(source);
-            return new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
+            await new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
+            try
+            {
```

Comment:
```suggestion
         try
         {
            await new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
   
```

Reply:
Isn't throwing the exception more close to the old behavior?

Reply:
I figure there are other things that might go wrong in the SaveImage method (external connection and stuff) and we would still cleanup even in the even of an error or?

Reply:
oh i see the confusion

Reply:
The old behavior would be remove the temp file after saving to metadata folder. If the save failed just throw, at least that was the old implementation in `BaseDynamicImageProvider`. This PR made that also available for `ItemImageProvider`. From current implementation in both use cases the source files are temp files so should be safe enough to remove, but we still want to throw the exception so that the task will properly fail.

Reply:
Ok i see that, but still in the case the SaveImage method throws, we still would want to delete the temp file or? So it should rather be:

```csharp
 	 try
	 {
	    await new ImageSaver(_configurationManager, _libraryMonitor, _fileSystem, _logger).SaveImage(item, fileStream, mimeType, type, imageIndex, saveLocallyWithMedia, cancellationToken);
            
      }
      catch (IOException ex)
      {
          _logger.LogError(ex, "Source file {Source} not found or in use, skip removing", source);
      }
      finally
      {
      	  if (File.Exists(source))
      	  {
      	  	      File.Delete(source);
      	  }
      }
```

If the delete fails, there aint much we can do anyway.